iotdb-operator/internal/controller/confignode_controller.go

```diff
@@ -0,0 +1,332 @@
+	envVars[envNum+1] = corev1.EnvVar{Name: "cn_seed_config_node", Value: val1}
+	envVars[envNum+2] = corev1.EnvVar{Name: "cn_internal_address", Value: val2}
+
+	if len(configNode.Spec.VolumeClaimTemplates) == 1 {
+		volumeClaimTemplates := make([]corev1.PersistentVolumeClaim, len(configNode.Spec.VolumeClaimTemplates))
+		pvcTemplate := configNode.Spec.VolumeClaimTemplates[0]
+		pvcName := pvcTemplate.Spec.Name
+		volumeClaimTemplates[0] = *r.constructPVCForConfigNode(configNode, pvcTemplate, pvcName)
+		statefulset := &appsv1.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      ConfigNodeName,
+				Namespace: configNode.Namespace,
+				Labels:    labels,
+			},
+			Spec: appsv1.StatefulSetSpec{
+				Replicas: &replicas,
+				Selector: &metav1.LabelSelector{
+					MatchLabels: labels,
+				},
+				ServiceName: ConfigNodeName + "-headless",
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: labels,
+					},
+					Spec: corev1.PodSpec{
+						Affinity: &corev1.Affinity{
+							PodAntiAffinity: &corev1.PodAntiAffinity{
+								RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+									{
+										LabelSelector: &metav1.LabelSelector{
+											MatchLabels: labels,
+										},
+										TopologyKey: "kubernetes.io/hostname",
+									},
+								},
+							},
+						},
+						Containers: []corev1.Container{
+							{
+								Name:            ConfigNodeName,
+								Image:           configNode.Spec.Image,
+								ImagePullPolicy: corev1.PullIfNotPresent,
+								Ports: []corev1.ContainerPort{
+									{Name: "internal", ContainerPort: 10710},
+									{Name: "consensus", ContainerPort: 10720},
+								},
+								Resources: corev1.ResourceRequirements{
+									Limits: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse(configNode.Spec.Resources.Limits.CPU),
+										corev1.ResourceMemory: resource.MustParse(configNode.Spec.Resources.Limits.Memory),
+									},
+									Requests: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse(configNode.Spec.Resources.Limits.CPU),
+										corev1.ResourceMemory: resource.MustParse(configNode.Spec.Resources.Limits.Memory),
+									},
+								},
+								Env: envVars,
+								VolumeMounts: []corev1.VolumeMount{
+									{Name: pvcName, MountPath: "/iotdb/data", SubPath: "data"},
+									{Name: pvcName, MountPath: "/iotdb/logs", SubPath: "logs"},
+									{Name: pvcName, MountPath: "/iotdb/ext", SubPath: "ext"},
+									{Name: pvcName, MountPath: "/iotdb/.env", SubPath: ".env"},
+									{Name: pvcName, MountPath: "/iotdb/activation", SubPath: "activation"},
+								},
+							},
+						},
+					},
+				},
+				VolumeClaimTemplates: volumeClaimTemplates,
+			},
+		}
+		controllerutil.SetControllerReference(configNode, statefulset, r.Scheme)
+		return statefulset
+	} else {
+		//todo add check
```

Comment:
if only one template is allowed, why not just make the field to accept a single value instead of a list ?